org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/AbstractAntlrGrammarGenerator.xtend

```diff
@@ -139,7 +139,7 @@ abstract class AbstractAntlrGrammarGenerator {
 	protected def compileTokens(Grammar it, AntlrOptions options) '''
 		�IF options.isBacktrackLexer�
 			tokens {
-				�FOR kw : allKeywords.sort.sortBy[-length]�
+				�FOR kw : keywordHelper.allKeywords.sort�
```

Comment:
The sort seems to be superfluous, as keywordHelper.allKeywords returns a sorted collection.